contentcuration/contentcuration/management/commands/set_tsvectors.py

```diff
@@ -0,0 +1,71 @@
+"""
+This command sets tsvector in title_description_search_vector field in batches.
+The batches are created on the basis of channel_id. This enables resumption. Also helps
+in cases of failure or memory overflow.
+"""
+import logging as logmodule
+
+from django.core.cache import cache
+from django.core.management.base import BaseCommand
+
+from contentcuration.models import Channel
+from contentcuration.models import ContentNode
+from contentcuration.models import POSTGRES_SEARCH_VECTOR
+
+
+logmodule.basicConfig(level=logmodule.INFO)
+logging = logmodule.getLogger(__name__)
+
+
+UPDATED_TS_VECTORS_CACHE_KEY = "tsvectors_updated_for_channel_ids"
+UPDATED_TS_VECTORS_FOR_NULL_CHANNEL_CACHE_KEY = "tsvectors_updated_for_null_channels"
+
+
+class Command(BaseCommand):
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--public",
+            action="store_true",
+            help="Set tsvector for only the public channel nodes instead of all nodes.",
+        )
+        parser.add_argument(
+            "--no-cache",
+            action="store_true",
+            help="Disables the cache. This updates all previously updated nodes.",
+        )
+
+    def handle(self, *args, **options):
+        if options["no_cache"]:
+            updated_channel_ids = []
+            do_update_nodes_with_null_channel_id = True
+        else:
+            updated_channel_ids = [] if cache.get(UPDATED_TS_VECTORS_CACHE_KEY) is None else cache.get(UPDATED_TS_VECTORS_CACHE_KEY)
```

Comment:
I thought about making it more efficient but couldn't come up with other alternatives. Any ideas sir?

Reply:
The database state should reflect what needs an updated ts_vector, and since "query performance isn't at a premium here", couldn't you query for published nodes that have a null ts_vector?

Reply:
Based on our decision about new table, data migration will be implemented in an entirely different way.